api/resources_portal/views/user.py

```diff
@@ -60,6 +61,14 @@ def create(self, request, *args, **kwargs):
 
         return response
 
+    def destroy(self, request, *args, **kwargs):
+        user = User.objects.get(pk=kwargs["pk"])
+        self.check_object_permissions(request, user)
+        user.deleted = True
+        user.save()
+
+        return Response(status=204)
```

Comment:
I think this logic should be done on the model instead of the view. However ideally we could use the `softdelete` package here so the behavior can be consistent.